web/src/pages/administration/CategoriesPage.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <v-card>
+    <v-card-text>
+      <div class="d-flex">
+        <v-text-field
+          v-model="search"
+          class="mb-4 mr-5"
+          label="Search"
+          density="compact"
+        />
+        <CategoryNewButton />
+      </div>
+
+      <v-data-table-server
+        v-model:items-per-page="perPage"
+        :search="search"
+        :items="categories"
+        :items-length="totalCount"
+        :page="page"
+        :loading="isLoading"
+        :headers="headers"
+        @update:options="loadItems"
+        @click:row="openItem"
+      >
+        <template #item.retention="{ item }">
+          {{ item.retention?.name }}
+        </template>
+      </v-data-table-server>
```

Comment:
Please consider making all table components their own thing as they are highly reusable and it keeps each individual component from becoming too complex.